packages/cadl-typescript/test/integration/arrayItemTypes.spec.ts

```diff
@@ -0,0 +1,93 @@
+import ArrayItemTypesClientFactory, {
+  ArrayItemTypesClient
+} from "./generated/array/itemTypes/src/index.js";
+import { assert } from "chai";
+import { matrix } from "../util/matrix.js";
+
+interface TypeDetail {
+  type: string;
+  defaultValue: any;
+  convertedToFn?: (_: any) => any;
+}
+
+const testedTypes: TypeDetail[] = [
+  {
+    type: "int32",
+    defaultValue: [1, 2]
+  },
+  {
+    type: "int64",
+    defaultValue: [Number.MAX_SAFE_INTEGER, Number.MIN_SAFE_INTEGER]
+  },
+  {
+    type: "boolean",
+    defaultValue: [true, false]
+  },
+  {
+    type: "string",
+    defaultValue: ["hello", ""]
+  },
+  {
+    type: "float32",
+    defaultValue: [42.42]
+  },
+  {
+    type: "datetime",
+    defaultValue: ["2022-08-26T18:38:00Z"]
+  },
+  {
+    type: "duration",
+    defaultValue: ["P123DT22H14M12.011S"]
+  },
+  {
+    type: "unknown",
+    defaultValue: [1, "hello", null]
+  },
+  {
+    type: "model",
+    defaultValue: [{ property: "hello" }, { property: "world" }]
+  }
+];
+describe("Array Item-Types Client", () => {
+  let client: ArrayItemTypesClient;
+
+  beforeEach(() => {
+    client = ArrayItemTypesClientFactory({
+      allowInsecureConnection: true
+    });
+  });
+
+  matrix([testedTypes], async (params: TypeDetail) => {
+    it(`should get a ${params.type} value`, async () => {
+      try {
```

Comment:
do we need to try catch? I think if something throws it will automatically fail won't it?

Reply:
It will skip the rest of testing but here we want to continue testing